### PR TITLE
Add simple smoke test for the main worker

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: cellfinder_napari/tests/test_cellfinder_napari.py [hunk 1]
```diff
@@ -7,51 +7,33 @@
...
-# def background_image():
-#    return load_stack(test_data_dir / "crop_planes" / "ch1")
-
-
-def test_detect_function(
-    make_napari_viewer,
-):  # , signal_image, background_image):
-    # Smoke test that running cell detection from the widget works
-
-    # Create viewer and add images
```
Comment (paddyroddy) [inline cellfinder_napari/tests/test_cellfinder_napari.py:40]: Is this part of the PR? I've seen this before, and potentially something to add to our developer conventions. To me, it looks like this is tidying up the tests and IMO should be part of a different PR

File: cellfinder_napari/tests/test_cellfinder_napari.py [hunk 2]
```diff
@@ -7,51 +7,33 @@
...
+    DataInputs,
+    DetectionInputs,
+    MiscInputs,
+)
+from cellfinder_napari.detect.thread_worker import Worker
+from cellfinder_napari.sample_data import load_sample
+
+
+def test_add_detect_widget(make_napari_viewer):
+    # Smoke test to check that adding detection widget works
```
Comment (paddyroddy) [inline cellfinder_napari/tests/test_cellfinder_napari.py:21]: I might be wrong, but it looks like this is a description of the test and, as such, should be a docstring
```suggestion
    """
    Smoke test to check that adding detection widget works
    """
```

**Review by paddyroddy — CHANGES_REQUESTED** [review]
Some comments/concerns (seen you've since made it draft as I was reviewing it)

File: cellfinder_napari/tests/test_cellfinder_napari.py (hunk 1 above)
Comment (dstansby) [reply cellfinder_napari/tests/test_cellfinder_napari.py:40]: Yes, tidying up the tests is part of this PR. A lot of the code I deleted is redundant because it's easy to get data via. the new `sample_data` function.

File: cellfinder_napari/tests/test_cellfinder_napari.py [hunk 3]
```diff
@@ -7,51 +7,33 @@
...
 
-    # Select images, and run detection
-    widget.reset_choices()
-    # widget.signal_image.value = signal_layer
-    # widget.background_image.value = background_layer
 
-    # TODO: work out how to run detection here whilst blocking execution.
-    # Currently calling widget() will start detection, but pytest will
-    # pass immediately because new processes are spawned to do the detection.
-    # Perhaps use https://pytest-qt.readthedocs.io/en/latest/signals.html ?
```
[the remark this answers is not in the document]
Comment (dstansby) [reply cellfinder_napari/tests/test_cellfinder_napari.py:57]: Yes, the new test I've added is directly checking detection.

**Comment by dstansby** [comment]
I switched this to draft since the tests are failing. This is because the new test requires https://github.com/brainglobe/cellfinder-core/pull/37 to be in a new release of `cellfinder-core.

File: cellfinder_napari/tests/test_cellfinder_napari.py (hunk 1 above)
Comment (paddyroddy) [reply cellfinder_napari/tests/test_cellfinder_napari.py]: Okay, I guess the title "Add ..." is a bit misleading

File: cellfinder_napari/tests/test_cellfinder_napari.py [hunk 4]
```diff
@@ -7,51 +7,33 @@
...
+from cellfinder_napari.sample_data import load_sample
+
+
+def test_add_detect_widget(make_napari_viewer):
+    # Smoke test to check that adding detection widget works
     viewer = make_napari_viewer()
-    # signal_layer = viewer.add_image(signal_image, name='signal')
-    # background_layer = viewer.add_image(background_image, name='background')
-
-    # Create widget
```
[the remark this answers is not in the document]
Comment (dstansby) [reply cellfinder_napari/tests/test_cellfinder_napari.py:45]: It seemed obvious to me what's going on from the code itself, but I can put this line back?

[the remark this answers is not in the document]
Comment (paddyroddy) [reply cellfinder_napari/tests/test_cellfinder_napari.py:45]: I don't know 🤷  don't see all the surrounding code on the PR view. If you think it's redundant, remove it

File: cellfinder_napari/detect/input_containers.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from typing import Optional
 
 import napari
+import numpy
```
Comment (paddyroddy) [inline cellfinder_napari/detect/input_containers.py:7]: A rare occasion where I've seen this without `np`

**Review by paddyroddy — APPROVED** [review]
